R/query-string.R

```diff
@@ -8,29 +8,36 @@ queryStringFilter <- function(req){
 #' @importFrom utils URLdecode
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == ""){
+  # Is there data in the request?
```

Comment:
The original motivation for this feature was to support browser <form>s which tend to use the key/value (querystring) format in a POST format. I think it's important to maintain support for that so browser forms can easily get data in. I do agree that most APIs would expect JSON, though.

I think your detection algorithm is reasonable, my comment was just about the name of the function. `parseQS` sounds like it only parses query strings. In reality, this `parseQS` function now parses either query strings or JSON. I think that's the right behavior, but the function name is now a misnomer. I was just hoping to bring the function name up to speed with what the current behavior of the function is.

I think the behavior you have now is right -- users should be able to POST query strings or JSON and have it just work.